scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtVersion.scala

```diff
@@ -2,7 +2,13 @@ package org.scalafmt.dynamic
 
 import scala.util.control.{NonFatal, NoStackTrace}
 
-case class ScalafmtVersion(major: Int, minor: Int, patch: Int, rc: Int) {
+case class ScalafmtVersion(
+    major: Int,
+    minor: Int,
+    patch: Int,
+    rc: Int,
+    snapshot: Boolean
+) {
   private val integerRepr: Int =
     major * 100 + minor * 10 + patch
```

Comment:
optional: if you replace with something like:
```
private val integerRepr: Int =
  major << 20 | minor << 15 | patch << 10 | rc << 5 | (snapshot == null)
```
can you then simplify comparison as:
```
def <(other: ScalafmtVersion): Boolean
  integerRepr < other.integerRepr
```